runner/src/server/services/uploadService.ts

```diff
@@ -56,34 +62,24 @@ export class UploadService {
     }
 
     const data = { headers: form.getHeaders(), payload: form };
-    const { res } = await post(`${config.documentUploadApiUrl}/v1/files`, data);
-    return this.parsedDocumentUploadResponse(res);
-  }
-
-  parsedDocumentUploadResponse(res: http.IncomingMessage) {
-    let error: string | undefined;
-    let location: string | undefined;
-
-    switch (res.statusCode) {
-      case 201:
-        location = res.headers.location;
-        break;
-      case 413:
-        error = 'The selected file for "%s" is too large';
-        break;
-      case 422:
-        error = 'The selected file for "%s" contained a virus';
-        break;
-      case 400:
-        error = "Invalid file type. Upload a PNG, JPG or PDF";
-        break;
-      default:
-        error = "There was an error uploading your file";
+    const { res, payload } = await post(
+      `${config.documentUploadApiUrl}/v1/files`,
+      data
+    );
+    if (res.statusCode === 201) {
+      return {
+        location: res.headers.location,
+        error: undefined,
+      };
     }
+    return this.parsedDocumentUploadResponse(payload);
+  }
 
+  parsedDocumentUploadResponse(error?: any) {
+    this.logger.info(["uploadService", "error"], error);
     return {
-      location,
-      error,
+      location: undefined,
+      error: ERRORS[error] ?? "There was an error uploading your file",
```

Comment:
you'll still need to use the switch, to maintain backwards compatibility.

I think probably what would make more sense is to just return the entire data object (response from wreck - I wish they had a better name for it), which is the object containing `res` and `payload`.

```
 const data = await post(`${config.documentUploadApiUrl}/v1/files`, data);
 // which is an object of { res, payload }
 const parsedResponse = this.parsedDocumentUploadResponse(data);
```


```.ts
parsedDocumentUploadResponse({ res, payload }) {
  const errorCodeFromApi = payload?.toString?.();
  let error;


  switch(res.statusCode) {
    case 201:
      //..
    case 422: 
      error = ERRORS[errorCodeFromApi] ?? ERRORS.virusError;
  }

  return { location, error }
}
```

since the case statement still relies on statusCode, we don't have to worry about doing an early return